### DISCUSSION TITLE
Clipboard API

### DISCUSSION THREAD

**Discussion by bettywolfy** [nesbox/TIC-80#2647] [Ideas]
I suggest an API for the Tic-80 to communicate with the device's clipboard.

```lua
local texts = clipboard(); -- table
trace(texts[#texts]); -- last text copied

clipboard("thanks for playing!"); -- adds a new item to the clipboard
```

**Comment by joshgoebel** [comment]
How would this concept even be relevant for a virtual console?

**Reply by bettywolfy** [reply]
example: use in cartridges with level editor, being able to share levels via code

**Reply by bettywolfy** [reply]
and I need it, otherwise why would I have asked for it?

**Comment by joshgoebel** [comment]
I see the potentia utiity.  Just kind of breaks the virtual console 4th wall in my opinion. :-)

**Reply by bettywolfy** [reply]
you're right :-(

**Comment by SRB2er** [comment]
i also think such a thing would be a cool addition to tic80
yes, its against the 'fantasy console' 4th wall, but it could have some nice versatility

for example (related to an editor like how  **eliaswolfy** mentioned ), you could make a small art tool, and be able to save 'files' as a string so it could be loaded later.
right now you can only really copy data (and its not even that easy to do) from console via trace, but a dedicated API would be greatly appreciated.

...an idea i just thought of while writing this, maybe we could have like a dedicated section of RAM that you could put this clipboard data in?
and you could, when needed, send this section of RAM to the clipboard. (with a set limit, obviously)
if you need to _get_ that data, you could read the current clipboard and send that data to said section of RAM.

something like the following:

```lua
-- get data from clipboard
getClip() --copies to RAM
-- maybe optional parameter [length] to get a set amount?

--send data to clipboard
setClip(512)
-- parameter [length] used to get certain amount
--this would also mean you couldn't have multiple bits of clip data in RAM, which i think fits the fantasy console idea.
```

thats just my take on such an idea though.
